shreesh_tripathi/assignment_two/RectMat.cu

```diff
@@ -0,0 +1,206 @@
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include<iomanip>
+#include<chrono>
+using namespace std;
+using namespace std::chrono;
+
+//Device code
+__global__ void AddN(float* a, float* b, 
+    float* c, int* ro, int* co){
+    int i, j, idx;
+    //assign a thread to each element (i,j)
+    i = blockIdx.x * blockDim.x + threadIdx.x;
+    j = blockIdx.y * blockDim.y + threadIdx.y;
+    
+    /*
+    Addressing logic:
+    Traverse i number of column lengths followed 
+    by an offset of j
+    */
+
+    idx = i * (*co) + j;
```

Comment:
This should be j * (*co) + i. 
2d arrays are stored in row major ordering, so you need to traverse j number of column lengths followed by offset of i.
Since i is your index in x dimension and j is the index in y dimension.